src/fu-util.c

```diff
@@ -1686,14 +1688,23 @@ fu_util_report_export(FuUtilPrivate *priv, gchar **values, GError **error)
 						     error);
 			if (sig == NULL)
 				return FALSE;
-			filename_sig = g_strdup_printf("%s.p7c", filename_json);
-			if (!g_file_set_contents(filename_sig, sig, -1, error))
-				return FALSE;
-			/* TRANSLATORS: the device ID is printed after this label */
-			g_print("%s %s\n",
-				_("Exported detached signature for offline report for device:"),
-				filename_sig);
+			sig_blob = g_bytes_new(sig, strlen(sig));
+			sig_img = fu_firmware_new_from_bytes(sig_blob);
+			fu_firmware_set_id(sig_img, "report.json.p7c");
+			fu_firmware_add_image(archive, sig_img);
 		}
+
+		/* save to local file */
+		fu_archive_firmware_set_format(FU_ARCHIVE_FIRMWARE(archive), FU_ARCHIVE_FORMAT_ZIP);
+		fu_archive_firmware_set_compression(FU_ARCHIVE_FIRMWARE(archive),
+						    FU_ARCHIVE_COMPRESSION_GZIP);
+		filename = g_strdup_printf("%s.zip", fwupd_device_get_id(dev));
```

Comment:
Shouldn't this be .gz since it's gzip?

Reply:
Nah, it's a bit of a weird misnomer -- it's a ZIP container format with MSZIP block compression. Maybe it makes sense to call the archive something else -- e.g. a `a45df35ac0e948ee180fe216a5f703f32dda163f.rzip` file looks less inviting to open and edit than a `a45df35ac0e948ee180fe216a5f703f32dda163f.zip` file..

Reply:
Sounds good

Reply:
Done. I used `.fwupdreport`